### PR TITLE
tekton task to run linting

### LINKED ISSUE #558: Create `Task` for running linting (markdownlint, …)
# Expected Behavior

We should be testing and releasing Pipelines with Pipelines.

We should have a `Task` (and related Resources) defined that we can use to:

- Run our "linting" (`markdownlint`, …)

## The Task

* Inputs:
  * git repo
* Outputs:
  * some kind of result info which can be passed to #535
* Parameters:  
  * some kind of gubernator endpoint info, will have to look into existing scripts to figure this out

# Actual Behavior

We currently use [prow](https://github.com/knative/test-infra/tree/master/ci/prow).

This is where the current markdown linting is implemented: [markdown_build_tests](https://github.com/knative/test-infra/blob/6972ae2803c107ef379a2e66c3efd33757f6ac0f/scripts/presubmit-tests.sh#L111)

* It uses https://github.com/markdownlint/markdownlint [to link the markdown files](https://github.com/knative/test-infra/blob/6972ae2803c107ef379a2e66c3efd33757f6ac0f/scripts/library.sh#L361)
* It uses https://github.com/raviqqe/liche [to lint the links in the markdown files](https://github.com/knative/test-infra/blob/6972ae2803c107ef379a2e66c3efd33757f6ac0f/scripts/library.sh#L351)

Note that these both only are applied to markdown files that have changed - we can leave that out of scope for this issue and apply the linting to all markdown files each run.

# Additional Info

* Actually triggering this is not in scope for this issue, it's fine to test manually (we'll be linking these together in a later issue)

/cc @bobcatfish

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This adds a Task to run `golangci-lint` on the repository.
This is part of the dogfooding effort on tektoncd/pipeline.

- This will move to `tektoncd/catalog` later on as it's meant to be a reusable task :angel: 
- This will be hooked-up to the CI once we work on #922

Closes #558 

cc @bobcatfish 
cc @chmouel I know you'll like it :angel: 

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ :no_good_man: ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ :no_good_man: ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by chmouel** [comment]
nice one, we may will have a need for a golangci.yaml !!

File: tekton/ci-lint.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+spec:
+  inputs:
+    params:
+    - name: package
+      description: package (and its children) under test
+      default: github.com/tektoncd/pipeline
+    - name: flags
+      description: flags to use for the test command
+    - name: version
+      default: golangci-lint version to use
```
Comment (houshengbo) [inline tekton/ci-lint.yaml]: Should be "description", not default.

Comment (vdemeester) [reply tekton/ci-lint.yaml]: Indeed :man_facepalming:

**Comment by dlorenc** [comment]
It looks like you want this to live in catalog as well, should it just go straight there?

**Comment by vdemeester** [comment]
@dlorenc yeah I started working on those (this PR and https://github.com/tektoncd/pipeline/pull/949) but I feel it's better to have them in the catalog in a slightly more generic way :angel: (and just "apply" those on the test cluster at first)

**Comment by vdemeester** [comment]
@dlorenc let's merge this like the unit test one for now, and we'll refine this from https://github.com/tektoncd/catalog/pull/43 and https://github.com/tektoncd/pipeline/issues/964

File: tekton/ci-lint.yaml (hunk 1 above)
Comment (bobcatfish) [reply tekton/ci-lint.yaml]: bump :D

File: tekton/ci-lint.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+metadata:
+  name: golangci-lint
+spec:
+  inputs:
+    params:
+    - name: package
+      description: package (and its children) under test
+      default: github.com/tektoncd/pipeline
+    - name: flags
+      description: flags to use for the test command
```
Comment (bobcatfish) [inline tekton/ci-lint.yaml]: i think this description might not be quite right? ("for the lint command"?)

Comment (vdemeester) [reply tekton/ci-lint.yaml]: :man_facepalming:

File: tekton/README.md [hunk 3]
```diff
@@ -13,6 +13,23 @@ TODO(#538): In #538 or #537 we will update
...
+  [`golangci-lint`](https://github.com/golangci/golangci-lint) to
+  validate the code based on common best practices.
+
+TODO(#922) & TODO(#860): Add the Pipeline and hook it up with Prow,
+for now all we have are `Tasks` which we can invoke individually by
+creating
+[`TaskRuns`](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md)
+and
+[`PipelineResources`](https://github.com/tektoncd/pipeline/blob/master/docs/resources.md).
+
```
Comment (bobcatfish) [inline tekton/README.md]: ty ty! 🎉

File: tekton/ci-lint.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+metadata:
+  name: golangci-lint
+spec:
+  inputs:
+    params:
+    - name: package
+      description: package (and its children) under test
+      default: github.com/tektoncd/pipeline
+    - name: flags
+      description: flags to use for the lint command
```
Comment (bobcatfish) [inline tekton/ci-lint.yaml]: silly question, but should flags be required for the lint command?

File: tekton/ci-lint-run.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+kind: TaskRun
+metadata:
+  name: golangci-lint-run
+spec:
+  taskRef:
+    name: golangci-lint
+  inputs:
+    params:
+    - name: flags
+      value: --verbose
```
Comment (bobcatfish) [inline tekton/ci-lint-run.yaml]: 😇 since this is the default we probably don't need it anymore? 😇

**Pull request merged**